src/lib/rules/validate-set-cookie-header/validate-set-cookie-header.ts

```diff
@@ -0,0 +1,226 @@
+        const headerName = 'set-cookie';
+
+        // Trim double quote
+        const unquote = (string: string): string => {
+            return string.replace(/(^")|("$)/g, '');
+        };
+
+        const normalizeAfterSplitByEqual = (splitResult: Array<string>): Array<string> => {
+            const [key, ...value] = splitResult;
+
+            return [normalizeString(key), unquote(value.join('='))];
+        };
+
+        // `set-cookie-string` parser based on algorithm used by a user agent defined in spec:
+        // https://tools.ietf.org/html/rfc6265#section-5.2.1
+        const parse = (setCookieValue: string) => {
+            const [nameValuePair, ...directivePairs] = setCookieValue.split(';');
+            const [cookieName, cookieValue] = normalizeAfterSplitByEqual(nameValuePair.split('='));
+
+            const cookie = {
+                name: cookieName,
+                value: cookieValue
+            };
+
+            if (directivePairs[directivePairs.length - 1] === '') {
+                throw new Error(`'${headerName}' header to set '${cookie.name}' has trailing ';'`);
+            }
+
+            directivePairs.forEach((part) => {
+                const [directiveKey, directiveValue] = normalizeAfterSplitByEqual(part.split('='));
+
+                if (cookie[directiveKey]) {
+                    throw new Error(`'${headerName}' header contains more than one ${directiveKey}.`);
+                }
+
+                cookie[directiveKey] = directiveValue || true;
+
+            });
+
+            return cookie;
+        };
+
+        const validASCII = (string: string): Boolean => {
+            return (/^[\x00-\x7F]+$/).test(string); // eslint-disable-line no-control-regex
+        };
+
+        // Validate cookie name or value string
+        // Ref: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie
+        const validString = (name: string, illegalChars: string): Boolean => {
+            const includesIllegalChars = illegalChars.split('').some((char) => {
+                return name.includes(char);
+            });
+            const includesWhiteSpace = (/\s/g).test(name);
+
+            return validASCII(name) && !includesIllegalChars && !includesWhiteSpace;
+        };
+
+        const validate = (fetchEnd: IFetchEnd) => {
+            const { element, resource, response }: { element: IAsyncHTMLElement, resource: string, response: IResponse } = fetchEnd;
+
+            // This check does not apply if URI starts with protocols others than http/https.
+            if (!hasRequiredProtocol(resource)) {
+                debug(`Check does not apply for URI: ${resource}`);
+
+                return;
+            }
+
+            const setCookieHeaders: Array<string> = response.headers && response.headers['set-cookie'];
+
+            if (!setCookieHeaders) {
+                return;
+            }
+
+            setCookieHeaders.forEach(async (setCookieValue) => {
```

Comment:
this method is too big. I think you can refactor it and move some of the checks to functions. i.e: `checkDateFormat` or `validateCookiePrefix`